src/sqlfluff/core/linter/linter.py

```diff
@@ -494,112 +495,129 @@ def lint_fix_parsed(
             ignore_buff = []
 
         save_tree = tree
-        for loop in range(loop_limit):
-            linter_logger.info(f"Linter loop {loop+1}/{loop_limit}")
-            changed = False
-
-            progress_bar_crawler = tqdm(
-                rule_set,
-                desc="lint by rules",
-                leave=False,
-                disable=progress_bar_configuration.disable_progress_bar,
-            )
-
-            for crawler in progress_bar_crawler:
-                progress_bar_crawler.set_description(f"rule {crawler.code}")
-
-                # fixes should be a dict {} with keys edit, delete, create
-                # delete is just a list of segments to delete
-                # edit and create are list of tuples. The first element is the
-                # "anchor", the segment to look for either to edit or to insert BEFORE.
-                # The second is the element to insert or create.
-                linting_errors, _, fixes, _ = crawler.crawl(
-                    tree,
-                    ignore_mask=ignore_buff,
-                    dialect=config.get("dialect_obj"),
-                    fname=fname,
-                    fix=fix,
-                    templated_file=templated_file,
+        for phase in ["main", "post"]:
```

Comment:
Use "Hide whitespace" to review the changes in this section, since it adds a new level of indentation.